iceoryx_hoofs/functional/include/iox/detail/storable_function.inl

```diff
@@ -176,13 +184,15 @@ inline constexpr void*
 storable_function<Capacity, signature<ReturnType, Args...>>::safeAlign(void* startAddress) noexcept
 {
     static_assert(is_storable<T>(), "type does not fit into storage");
-    // AXIVION DISABLE STYLE AutosarC++19_03-A5.2.4, AutosarC++19_03-M5.2.9 : Cast required for low level pointer alignment
+    // AXIVION DISABLE STYLE AutosarC++19_03-A5.2.4 : Cast required for low level pointer alignment
```

Comment:
Seriously?

Reply:
I mean if you have a suggestion I will add it

Reply:
What do you think about the following?

```suggestion
    // AXIVION DISABLE STYLE AutosarC++19_03-A5.2.4 : Required for low level pointer alignment, cast is safe as 'T' is used for alignment and type reification
```

Reply:
@mossmaurice I'm hesitant to add this justification since it is wrong from my point of view. I could call `safeAlign<bool>` from somewhere in the class which would break the code and contradict the proposed justification. It is not safe to use `T` but required and the context of the call makes it safe.

Please propose something better or accept the current justification which did not change in this PR.

Reply:
Me as an innocent bystander. The method is called `safeAlign` and it aligns a given memory position. So encapsulating this 3 code lines with reinterpret_cast increases safe usage inside the abstraction.
The `align` method is already tested, which everything is in our code base, so using reinterpret_cast inside of this is the safest approach.

But all of this is easily readable directly inside those 3 lines.

@mossmaurice 

> cast is safe as 'T' is used for alignment and type reification

But this does not state why it is safe, it states for what T is used.

I would suggest to leave the comment as it is and get this merged. The justification format: "why is it required & why is it safe" should provide more of a guideline but is not carved in stone and one has to always see the context. With the 3 easily readable code lines the context is clear and the "why is it required & why is it safe" becomes obvious.

Reply:
@mossmaurice I'm merging this now. If you think that any of the justifications needs improvements feel free to open a PR

Reply:
@elfenpiff Alright, makes sense. Closing this thread.